### PR TITLE
SPIN-1768 Manual Trigger Fast Property.

### PR DESCRIPTION
**Pull request by zanthrash**
Adds the ability to edit multiple Fast Properties key/value pairs for all
Fast Property stage when manually triggering a pipeline.

Also adds the ability to create a new Fast Property stage from an existing
one.

Cleaned up some whitespace.

@anotherchrisberry PTAL

### CODE REVIEW

File: app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js [hunk 1]
```diff
@@ -87,11 +98,25 @@ module.exports = angular.module('spinnaker.core.delivery.manualPipelineExecution
       if (pipeline.parameterConfig && pipeline.parameterConfig.length) {
         selectedTrigger.parameters = this.parameters;
       }
-      $uibModalInstance.close(command);
+
+      if(this.pipelineIsDirty()) {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js]: does the linter complain about the non-space between `if` and `(`?

File: app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.html [hunk 2]
```diff
@@ -97,7 +100,32 @@ <h3 ng-if="!vm.command.pipeline">Select Pipeline</h3>
           </div>
         </div>
       </div>
+
+      <div ng-if="vm.hasStageOf('createProperty')">
+
+        <p class="manual-execution-parameters-description">This pipeline is contains <b>Persisted Property Stages</b>. Please enter the properties and values for  each stage below.</p>
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.html]: s/is contains/contains

File: app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js [hunk 3]
```diff
@@ -87,11 +98,25 @@ module.exports = angular.module('spinnaker.core.delivery.manualPipelineExecution
...
+        pipelineConfigService.savePipeline(pipeline)
+          .then(() => $uibModalInstance.close(command) );
+      } else {
+        $uibModalInstance.close(command);
+      }
+
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js]: How does this happen? Is it that we're saving the rollback values in the pipeline config or something?

File: app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js (hunk 1 above)
Comment (zanthrash) [inline app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js]: No.

File: app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.html (hunk 2 above)
Comment (zanthrash) [inline app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.html]: good catch

File: app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js (hunk 3 above)
Comment (zanthrash) [inline app/scripts/modules/core/delivery/manualExecution/manualPipelineExecution.controller.js]: This is just allowing the user to change the FP stages before the pipeline is executed.

**Comment by anotherchrisberry** [comment]
LGTM - needs a rebase

**Pull request merged**